pkg/crc/machine/kubeconfig_test.go

```diff
@@ -59,3 +61,75 @@ func TestUpdateUserCaAndKeyToKubeconfig(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, "dummycert", userClientCA)
 }
+
+func createTempKubeConfig(config *api.Config) (string, error) {
+	tempFile, err := os.CreateTemp("", "kubeconfig-")
+	if err != nil {
+		return "", err
+	}
+	path := tempFile.Name()
+
+	err = clientcmd.WriteToFile(*config, path)
+	if err != nil {
+		os.Remove(path)
+		return "", err
+	}
+
+	return path, nil
+}
+
+func Test_mergeKubeConfigFile(t *testing.T) {
+	// Define two simple configurations
+	primaryConfig := api.NewConfig()
+	primaryConfig.Clusters["primary-cluster"] = &api.Cluster{Server: "https://primary.example.com"}
+	primaryConfig.AuthInfos["primary-user"] = &api.AuthInfo{Token: "primary-token"}
+	primaryConfig.Contexts["primary-context"] = &api.Context{Cluster: "primary-cluster", AuthInfo: "primary-user"}
+	primaryConfig.CurrentContext = "primary-context"
+
+	secondaryConfig := api.NewConfig()
+	secondaryConfig.Clusters["secondary-cluster"] = &api.Cluster{Server: "https://secondary.example.com"}
+	secondaryConfig.AuthInfos["secondary-user"] = &api.AuthInfo{Token: "secondary-token"}
+	secondaryConfig.Contexts["secondary-context"] = &api.Context{Cluster: "secondary-cluster", AuthInfo: "secondary-user"}
+	secondaryConfig.CurrentContext = "secondary-context"
+
+	// Create temporary kubeconfig files for the primary and secondary configurations
+	primaryConfigPath, err := createTempKubeConfig(primaryConfig)
+	if err != nil {
+		t.Fatalf("failed to create temporary kubeconfig file: %v", err)
+	}
+	defer os.Remove(primaryConfigPath)
+
+	secondaryConfigPath, err := createTempKubeConfig(secondaryConfig)
+	if err != nil {
+		t.Fatalf("failed to create temporary kubeconfig file: %v", err)
+	}
+	defer os.Remove(secondaryConfigPath)
+	err = mergeKubeConfigFile(secondaryConfigPath, primaryConfigPath)
+	if err != nil {
+		t.Fatalf("failed to modify kubeconfig: %v", err)
+	}
+
+	// Load the modified kubeconfig to ensure it was merged correctly
+	mergedConfig, err := clientcmd.LoadFromFile(primaryConfigPath)
+	if err != nil {
+		t.Fatalf("failed to load merged kubeconfig: %v", err)
+	}
```

Comment:
can use `assert.NoError` for the error checks, https://pkg.go.dev/github.com/stretchr/testify/assert#NoError